<h1>Palmer Penguin Data Analytics<h1>
<hr>
<h3>Analyst : Ana Indonilla<h3>


In [288]:
import numpy as np
import pandas as pd
import warnings
warnings.filterwarnings('ignore')

In [289]:
penguins = pd.read_csv(r"C:\Users\Admin\Downloads\penguins.csv")
penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen            39.1           18.7              181.0   
1       Adelie  Torgersen            39.5           17.4              186.0   
2       Adelie  Torgersen            40.3           18.0              195.0   
3       Adelie  Torgersen             NaN            NaN                NaN   
4       Adelie  Torgersen            36.7           19.3              193.0   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream            55.8           19.8              207.0   
340  Chinstrap      Dream            43.5           18.1              202.0   
341  Chinstrap      Dream            49.6           18.2              193.0   
342  Chinstrap      Dream            50.8           19.0              210.0   
343  Chinstrap      Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[344 rows x 8 columns]

<h1>Descriptive Analytics<h1>

In [290]:
penguins.describe()

bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g  \
count      342.000000     342.000000         342.000000   342.000000   
mean        43.921930      17.151170         200.915205  4201.754386   
std          5.459584       1.974793          14.061714   801.954536   
min         32.100000      13.100000         172.000000  2700.000000   
25%         39.225000      15.600000         190.000000  3550.000000   
50%         44.450000      17.300000         197.000000  4050.000000   
75%         48.500000      18.700000         213.000000  4750.000000   
max         59.600000      21.500000         231.000000  6300.000000   

              year  
count   344.000000  
mean   2008.029070  
std       0.818356  
min    2007.000000  
25%    2007.000000  
50%    2008.000000  
75%    2009.000000  
max    2009.000000

In [291]:
penguins.describe().T

count         mean         std     min       25%      50%  \
bill_length_mm     342.0    43.921930    5.459584    32.1    39.225    44.45   
bill_depth_mm      342.0    17.151170    1.974793    13.1    15.600    17.30   
flipper_length_mm  342.0   200.915205   14.061714   172.0   190.000   197.00   
body_mass_g        342.0  4201.754386  801.954536  2700.0  3550.000  4050.00   
year               344.0  2008.029070    0.818356  2007.0  2007.000  2008.00   

                      75%     max  
bill_length_mm       48.5    59.6  
bill_depth_mm        18.7    21.5  
flipper_length_mm   213.0   231.0  
body_mass_g        4750.0  6300.0  
year               2009.0  2009.0

<h1>Cleaning Data<h1>

In [292]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

<h1> Data Frame</h1>

In [293]:
penguins['species'].unique()

array(['Adelie', 'Gentoo', 'Chinstrap'], dtype=object)

In [294]:
penguins['species'].value_counts()

species
Adelie       152
Gentoo       124
Chinstrap     68
Name: count, dtype: int64

In [295]:
adelie_df = penguins[penguins['species']=="Adelie"]
adelie_df

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0    Adelie  Torgersen            39.1           18.7              181.0   
1    Adelie  Torgersen            39.5           17.4              186.0   
2    Adelie  Torgersen            40.3           18.0              195.0   
3    Adelie  Torgersen             NaN            NaN                NaN   
4    Adelie  Torgersen            36.7           19.3              193.0   
..      ...        ...             ...            ...                ...   
147  Adelie      Dream            36.6           18.4              184.0   
148  Adelie      Dream            36.0           17.8              195.0   
149  Adelie      Dream            37.8           18.1              193.0   
150  Adelie      Dream            36.0           17.1              187.0   
151  Adelie      Dream            41.5           18.5              201.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
147       3475.0  female  2009  
148       3450.0  female  2009  
149       3750.0    male  2009  
150       3700.0  female  2009  
151       4000.0    male  2009  

[152 rows x 8 columns]

In [296]:
gentoo_df = penguins[penguins['species']=="Gentoo"]
gentoo_df

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
152  Gentoo  Biscoe            46.1           13.2              211.0   
153  Gentoo  Biscoe            50.0           16.3              230.0   
154  Gentoo  Biscoe            48.7           14.1              210.0   
155  Gentoo  Biscoe            50.0           15.2              218.0   
156  Gentoo  Biscoe            47.6           14.5              215.0   
..      ...     ...             ...            ...                ...   
271  Gentoo  Biscoe             NaN            NaN                NaN   
272  Gentoo  Biscoe            46.8           14.3              215.0   
273  Gentoo  Biscoe            50.4           15.7              222.0   
274  Gentoo  Biscoe            45.2           14.8              212.0   
275  Gentoo  Biscoe            49.9           16.1              213.0   

     body_mass_g     sex  year  
152       4500.0  female  2007  
153       5700.0    male  2007  
154       4450.0  female  2007  
155       5700.0    male  2007  
156       5400.0    male  2007  
..           ...     ...   ...  
271          NaN     NaN  2009  
272       4850.0  female  2009  
273       5750.0    male  2009  
274       5200.0  female  2009  
275       5400.0    male  2009  

[124 rows x 8 columns]

In [297]:
chinstrap_df = penguins[penguins['species']=="Chinstrap"]
chinstrap_df

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
276  Chinstrap  Dream            46.5           17.9              192.0   
277  Chinstrap  Dream            50.0           19.5              196.0   
278  Chinstrap  Dream            51.3           19.2              193.0   
279  Chinstrap  Dream            45.4           18.7              188.0   
280  Chinstrap  Dream            52.7           19.8              197.0   
..         ...    ...             ...            ...                ...   
339  Chinstrap  Dream            55.8           19.8              207.0   
340  Chinstrap  Dream            43.5           18.1              202.0   
341  Chinstrap  Dream            49.6           18.2              193.0   
342  Chinstrap  Dream            50.8           19.0              210.0   
343  Chinstrap  Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
276       3500.0  female  2007  
277       3900.0    male  2007  
278       3650.0    male  2007  
279       3525.0  female  2007  
280       3725.0    male  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[68 rows x 8 columns]

In [298]:
penguins[penguins["body_mass_g"] >= 4000]

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
7       Adelie  Torgersen            39.2           19.6              195.0   
9       Adelie  Torgersen            42.0           20.2              190.0   
14      Adelie  Torgersen            34.6           21.1              198.0   
17      Adelie  Torgersen            42.5           20.7              197.0   
19      Adelie  Torgersen            46.0           21.5              194.0   
..         ...        ...             ...            ...                ...   
323  Chinstrap      Dream            49.0           19.6              212.0   
329  Chinstrap      Dream            50.7           19.7              203.0   
333  Chinstrap      Dream            49.3           19.9              203.0   
339  Chinstrap      Dream            55.8           19.8              207.0   
342  Chinstrap      Dream            50.8           19.0              210.0   

     body_mass_g   sex  year  
7         4675.0  male  2007  
9         4250.0   NaN  2007  
14        4400.0  male  2007  
17        4500.0  male  2007  
19        4200.0  male  2007  
..           ...   ...   ...  
323       4300.0  male  2009  
329       4050.0  male  2009  
333       4050.0  male  2009  
339       4000.0  male  2009  
342       4100.0  male  2009  

[177 rows x 8 columns]

In [299]:
penguins[(penguins['species']=="Chinstrap") & (penguins["body_mass_g"] >= 4000)]

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
284  Chinstrap  Dream            46.0           18.9              195.0   
289  Chinstrap  Dream            52.0           18.1              201.0   
291  Chinstrap  Dream            50.5           19.6              201.0   
295  Chinstrap  Dream            49.2           18.2              195.0   
301  Chinstrap  Dream            52.0           19.0              197.0   
305  Chinstrap  Dream            52.8           20.0              205.0   
307  Chinstrap  Dream            54.2           20.8              201.0   
309  Chinstrap  Dream            51.0           18.8              203.0   
313  Chinstrap  Dream            52.0           20.7              210.0   
315  Chinstrap  Dream            53.5           19.9              205.0   
321  Chinstrap  Dream            50.8           18.5              201.0   
323  Chinstrap  Dream            49.0           19.6              212.0   
329  Chinstrap  Dream            50.7           19.7              203.0   
333  Chinstrap  Dream            49.3           19.9              203.0   
339  Chinstrap  Dream            55.8           19.8              207.0   
342  Chinstrap  Dream            50.8           19.0              210.0   

     body_mass_g     sex  year  
284       4150.0  female  2007  
289       4050.0    male  2007  
291       4050.0    male  2007  
295       4400.0    male  2007  
301       4150.0    male  2007  
305       4550.0    male  2008  
307       4300.0    male  2008  
309       4100.0    male  2008  
313       4800.0    male  2008  
315       4500.0    male  2008  
321       4450.0    male  2009  
323       4300.0    male  2009  
329       4050.0    male  2009  
333       4050.0    male  2009  
339       4000.0    male  2009  
342       4100.0    male  2009

In [300]:
penguins[(penguins['species']=="Chinstrap") & (penguins["body_mass_g"] >= 4000)].info()

<class 'pandas.core.frame.DataFrame'>
Index: 16 entries, 284 to 342
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            16 non-null     object 
 1   island             16 non-null     object 
 2   bill_length_mm     16 non-null     float64
 3   bill_depth_mm      16 non-null     float64
 4   flipper_length_mm  16 non-null     float64
 5   body_mass_g        16 non-null     float64
 6   sex                16 non-null     object 
 7   year               16 non-null     int64  
dtypes: float64(4), int64(1), object(3)
memory usage: 1.1+ KB


<h1>Filling up Null</h1>

In [301]:
penguins['bill_length_mm'] = penguins['bill_length_mm'].fillna(penguins['bill_length_mm'].mean())
penguins['sex'] = penguins['sex'].fillna(penguins['sex'].mode()[0])
penguins['bill_depth_mm'] = penguins['bill_depth_mm'].fillna(penguins['bill_depth_mm'].mean())
penguins['flipper_length_mm'] = penguins['flipper_length_mm'].fillna(penguins['flipper_length_mm'].mean())
penguins['body_mass_g'] = penguins['body_mass_g'].fillna(penguins['body_mass_g'].mean())

penguins.isnull().sum()

species              0
island               0
bill_length_mm       0
bill_depth_mm        0
flipper_length_mm    0
body_mass_g          0
sex                  0
year                 0
dtype: int64

<h1> How to Change Data Types of a Column

In [302]:
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            344 non-null    object 
 1   island             344 non-null    object 
 2   bill_length_mm     344 non-null    float64
 3   bill_depth_mm      344 non-null    float64
 4   flipper_length_mm  344 non-null    float64
 5   body_mass_g        344 non-null    float64
 6   sex                344 non-null    object 
 7   year               344 non-null    int64  
dtypes: float64(4), int64(1), object(3)
memory usage: 21.6+ KB


In [303]:
penguins['species'] = penguins['species'].astype('category')
penguins['island'] = penguins['island'].astype('category')
penguins['sex'] = penguins['sex'].astype('category')

penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   species            344 non-null    category
 1   island             344 non-null    category
 2   bill_length_mm     344 non-null    float64 
 3   bill_depth_mm      344 non-null    float64 
 4   flipper_length_mm  344 non-null    float64 
 5   body_mass_g        344 non-null    float64 
 6   sex                344 non-null    category
 7   year               344 non-null    int64   
dtypes: category(3), float64(4), int64(1)
memory usage: 15.0 KB


<h1> Data analysis (7 data) </h1>

<h3>1. How many female gentoo is there?</h3>

In [304]:
gentoo_df[gentoo_df["sex"] == 'female']

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
152  Gentoo  Biscoe            46.1           13.2              211.0   
154  Gentoo  Biscoe            48.7           14.1              210.0   
157  Gentoo  Biscoe            46.5           13.5              210.0   
158  Gentoo  Biscoe            45.4           14.6              211.0   
160  Gentoo  Biscoe            43.3           13.4              209.0   
162  Gentoo  Biscoe            40.9           13.7              214.0   
164  Gentoo  Biscoe            45.5           13.7              214.0   
166  Gentoo  Biscoe            45.8           14.6              210.0   
168  Gentoo  Biscoe            42.0           13.5              210.0   
170  Gentoo  Biscoe            46.2           14.5              209.0   
173  Gentoo  Biscoe            45.1           14.5              215.0   
174  Gentoo  Biscoe            46.5           14.5              213.0   
176  Gentoo  Biscoe            42.9           13.1              215.0   
180  Gentoo  Biscoe            48.2           14.3              210.0   
183  Gentoo  Biscoe            42.8           14.2              209.0   
184  Gentoo  Biscoe            45.1           14.5              207.0   
186  Gentoo  Biscoe            49.1           14.8              220.0   
188  Gentoo  Biscoe            42.6           13.7              213.0   
190  Gentoo  Biscoe            44.0           13.6              208.0   
192  Gentoo  Biscoe            42.7           13.7              208.0   
194  Gentoo  Biscoe            45.3           13.7              210.0   
197  Gentoo  Biscoe            43.6           13.9              217.0   
198  Gentoo  Biscoe            45.5           13.9              210.0   
200  Gentoo  Biscoe            44.9           13.3              213.0   
202  Gentoo  Biscoe            46.6           14.2              210.0   
204  Gentoo  Biscoe            45.1           14.4              210.0   
206  Gentoo  Biscoe            46.5           14.4              217.0   
208  Gentoo  Biscoe            43.8           13.9              208.0   
210  Gentoo  Biscoe            43.2           14.5              208.0   
212  Gentoo  Biscoe            45.3           13.8              208.0   
214  Gentoo  Biscoe            45.7           13.9              214.0   
216  Gentoo  Biscoe            45.8           14.2              219.0   
220  Gentoo  Biscoe            43.5           14.2              220.0   
222  Gentoo  Biscoe            47.7           15.0              216.0   
225  Gentoo  Biscoe            46.5           14.8              217.0   
226  Gentoo  Biscoe            46.4           15.0              216.0   
228  Gentoo  Biscoe            47.5           14.2              209.0   
230  Gentoo  Biscoe            45.2           13.8              215.0   
232  Gentoo  Biscoe            49.1           14.5              212.0   
234  Gentoo  Biscoe            47.4           14.6              212.0   
236  Gentoo  Biscoe            44.9           13.8              212.0   
238  Gentoo  Biscoe            43.4           14.4              218.0   
240  Gentoo  Biscoe            47.5           14.0              212.0   
242  Gentoo  Biscoe            47.5           15.0              218.0   
244  Gentoo  Biscoe            45.5           14.5              212.0   
246  Gentoo  Biscoe            44.5           14.7              214.0   
249  Gentoo  Biscoe            46.9           14.6              222.0   
250  Gentoo  Biscoe            48.4           14.4              203.0   
252  Gentoo  Biscoe            48.5           15.0              219.0   
254  Gentoo  Biscoe            47.2           15.5              215.0   
258  Gentoo  Biscoe            41.7           14.7              210.0   
260  Gentoo  Biscoe            43.3           14.0              208.0   
262  Gentoo  Biscoe            50.5           15.2              216.0   
264  Gentoo  Biscoe            43.5           15.2            

In [305]:
gentoo_df[gentoo_df["sex"] == 'female'].info()

<class 'pandas.core.frame.DataFrame'>
Index: 58 entries, 152 to 274
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            58 non-null     object 
 1   island             58 non-null     object 
 2   bill_length_mm     58 non-null     float64
 3   bill_depth_mm      58 non-null     float64
 4   flipper_length_mm  58 non-null     float64
 5   body_mass_g        58 non-null     float64
 6   sex                58 non-null     object 
 7   year               58 non-null     int64  
dtypes: float64(4), int64(1), object(3)
memory usage: 4.1+ KB


<h1>Insight</h1>
<h3>Out of all the gentoo, there's only a total of 58 female.</h3>

<h3>2. How many chinstrap is below 3000g? </h3>

In [306]:
chinstrap_df[chinstrap_df["body_mass_g"] <= 3000]

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
298  Chinstrap  Dream            43.2           16.6              187.0   
314  Chinstrap  Dream            46.9           16.6              192.0   

     body_mass_g     sex  year  
298       2900.0  female  2007  
314       2700.0  female  2008

In [307]:
chinstrap_df[chinstrap_df["body_mass_g"] <= 3000].info()

<class 'pandas.core.frame.DataFrame'>
Index: 2 entries, 298 to 314
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            2 non-null      object 
 1   island             2 non-null      object 
 2   bill_length_mm     2 non-null      float64
 3   bill_depth_mm      2 non-null      float64
 4   flipper_length_mm  2 non-null      float64
 5   body_mass_g        2 non-null      float64
 6   sex                2 non-null      object 
 7   year               2 non-null      int64  
dtypes: float64(4), int64(1), object(3)
memory usage: 144.0+ bytes


In [308]:
chinstrap_df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 68 entries, 276 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            68 non-null     object 
 1   island             68 non-null     object 
 2   bill_length_mm     68 non-null     float64
 3   bill_depth_mm      68 non-null     float64
 4   flipper_length_mm  68 non-null     float64
 5   body_mass_g        68 non-null     float64
 6   sex                68 non-null     object 
 7   year               68 non-null     int64  
dtypes: float64(4), int64(1), object(3)
memory usage: 4.8+ KB


<h1>Insights</h1>
<h3> Only 2 out of 68 chinstrap weigh below 3000g</h3>

<H3>3. What species has the most male?</H3>

In [309]:
chinstrap_df['sex'].value_counts()

sex
female    34
male      34
Name: count, dtype: int64

In [310]:
gentoo_df['sex'].value_counts()

sex
male      61
female    58
Name: count, dtype: int64

In [311]:
adelie_df['sex'].value_counts()

sex
male      73
female    73
Name: count, dtype: int64

<h1>Insights</h1>
<h3>From the given data, Adelie showed that it has the most male out of the three species.</h3>

<h3>4. What is the average body mass of each species?

In [312]:
gentoo_df['body_mass_g'].mean()

np.float64(5076.016260162602)

In [313]:
adelie_df['body_mass_g'].mean()

np.float64(3700.662251655629)

In [314]:
chinstrap_df['body_mass_g'].mean()

np.float64(3733.0882352941176)

<h1>Insights</h1>
<h3>From the data collected the average weight of each species is as follows:</h3>
<h4>-gentoo (average in g: 5076.016 )</h4>
<h4>-adelie (average in g: 3700.662 )</h4>
<h4>-chinstrap (average in g: 3733.088 ) </h4>

<H3>5. How many adelie data was recorded in year 2009?</H3>

In [315]:
adelie_df['year'].value_counts()

year
2009    52
2007    50
2008    50
Name: count, dtype: int64

<h1>Insights</h1>
<h3>A total of 52 entries was made in 2009</h3>

<h3>6. What is the total of the data and how many males and females are there?</h3>

In [316]:
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   species            344 non-null    category
 1   island             344 non-null    category
 2   bill_length_mm     344 non-null    float64 
 3   bill_depth_mm      344 non-null    float64 
 4   flipper_length_mm  344 non-null    float64 
 5   body_mass_g        344 non-null    float64 
 6   sex                344 non-null    category
 7   year               344 non-null    int64   
dtypes: category(3), float64(4), int64(1)
memory usage: 15.0 KB


In [317]:
penguins['sex'].value_counts()

sex
male      179
female    165
Name: count, dtype: int64

<h1>Insights</h1>
<h3>There are 179 males and 165 females with a total of 344 data entries.</h3>

<h3>7. What is the average flipper lenght of a gentoo?</h3>

In [318]:
gentoo_df['flipper_length_mm'].mean()

np.float64(217.1869918699187)

<h1>Insights</h1>
<h3>From the given data, the average flipper lenght of the gento in mm is 217.1869918699187</h3>

<h1> Data Aggregation </h1>

<h3>How many species are there per island</h3>

In [319]:
penguins.groupby(['species','island']).count()

bill_length_mm  bill_depth_mm  flipper_length_mm  \
species   island                                                        
Adelie    Biscoe                 44             44                 44   
          Dream                  56             56                 56   
          Torgersen              52             52                 52   
Chinstrap Biscoe                  0              0                  0   
          Dream                  68             68                 68   
          Torgersen               0              0                  0   
Gentoo    Biscoe                124            124                124   
          Dream                   0              0                  0   
          Torgersen               0              0                  0   

                     body_mass_g  sex  year  
species   island                             
Adelie    Biscoe              44   44    44  
          Dream               56   56    56  
          Torgersen           52   52    52  
Chinstrap Biscoe               0    0     0  
          Dream               68   68    68  
          Torgersen            0    0     0  
Gentoo    Biscoe             124  124   124  
          Dream                0    0     0  
          Torgersen            0    0     0

In [320]:
penguins.groupby(['species','island','sex']).size()

species    island     sex   
Adelie     Biscoe     female    22
                      male      22
           Dream      female    27
                      male      29
           Torgersen  female    24
                      male      28
Chinstrap  Biscoe     female     0
                      male       0
           Dream      female    34
                      male      34
           Torgersen  female     0
                      male       0
Gentoo     Biscoe     female    58
                      male      66
           Dream      female     0
                      male       0
           Torgersen  female     0
                      male       0
dtype: int64

<h3>Show the collected data of all the species in each year </h3>

In [321]:
penguins.groupby(['species','year']).size()

species    year
Adelie     2007    50
           2008    50
           2009    52
Chinstrap  2007    26
           2008    18
           2009    24
Gentoo     2007    34
           2008    46
           2009    44
dtype: int64

In [322]:
penguins.groupby(adelie_df['year']).size()

year
2007.0    50
2008.0    50
2009.0    52
dtype: int64

In [323]:
penguins.groupby(['year','sex','island'])['body_mass_g'].mean()

year  sex     island   
2007  female  Biscoe       4345.238095
              Dream        3446.590909
              Torgersen    3475.000000
      male    Biscoe       5102.173913
              Dream        3902.083333
              Torgersen    3991.812865
2008  female  Biscoe       4225.806452
              Dream        3444.117647
              Torgersen    3518.750000
      male    Biscoe       5006.060606
              Dream        4114.705882
              Torgersen    4193.750000
2009  female  Biscoe       4403.571429
              Dream        3447.727273
              Torgersen    3193.750000
      male    Biscoe       5114.898575
              Dream        3935.227273
              Torgersen    3784.375000
Name: body_mass_g, dtype: float64

<h1>8. How many Adelie are recorded in each island?<h1>


In [324]:
penguins.groupby(['species', 'island']).size()['Adelie']


island
Biscoe       44
Dream        56
Torgersen    52
dtype: int64

<h1>Insights</h1>
<h3> In Biscoe, there are 44 data recorded about Adelie. As for the Dream Island there are 56 and 52 in Torgersen </h3>

<h1>9. What is the average body mass of Gentoo with their sex in consideration?<h1>

In [325]:
gentoo_df.groupby(['species', 'sex'])['body_mass_g'].mean()


species  sex   
Gentoo   female    4679.741379
         male      5484.836066
Name: body_mass_g, dtype: float64

<h1>Insights</h1>
<h3>The average body mass of a female gentoo is 4679.74g while male gentoo is 5484.84g </h3>

<h1>10. On which island are the most male and females found, and which species do they belong to?

In [326]:
penguins.groupby(['species','island','sex']).size()

species    island     sex   
Adelie     Biscoe     female    22
                      male      22
           Dream      female    27
                      male      29
           Torgersen  female    24
                      male      28
Chinstrap  Biscoe     female     0
                      male       0
           Dream      female    34
                      male      34
           Torgersen  female     0
                      male       0
Gentoo     Biscoe     female    58
                      male      66
           Dream      female     0
                      male       0
           Torgersen  female     0
                      male       0
dtype: int64

<h1> Insights </h1>
<h3> The highest number of males and females are found on Biscoe Island, and they belong to the gentoo species.</h3>

<h1>11. Which year has the highest recorded number of adelie?</h3>

In [327]:
penguins[penguins['species'] == 'Adelie'].groupby(['species','year','sex']).size()

species    year  sex   
Adelie     2007  female    22
                 male      28
           2008  female    25
                 male      25
           2009  female    26
                 male      26
Chinstrap  2007  female     0
                 male       0
           2008  female     0
                 male       0
           2009  female     0
                 male       0
Gentoo     2007  female     0
                 male       0
           2008  female     0
                 male       0
           2009  female     0
                 male       0
dtype: int64

<h1> Insights </h1>
<h3> The year 2009 has the most number of adelie</h3>

<h1>12. How many data of chinstrap was recorded in 2007? <h1>

In [328]:
chinstrap_df.groupby(['species', 'year']).size()

species    year
Chinstrap  2007    26
           2008    18
           2009    24
dtype: int64

<h1>Insights</h1>
<h3>From the data given, a total of 24 data entries were recorded in 2009</h3>